### PR TITLE
[Merged by Bors] - feat(CategoryTheory): Split equalizers

### PR DESCRIPTION
**Pull request by mckoen**
Defines what it means for a triple of morphisms `f g : X ⟶ Y`, `ι : W ⟶ X` to be a split equalizer. In addition, shows that every split equalizer is an equalizer and absolute.

This dualises everything in [Mathlib.CategoryTheory.Limits.Shapes.SplitCoequalizer](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/Limits/Shapes/SplitCoequalizer.html).

This contribution was created as part of the AIM workshop "Formalizing Algebraic Geometry" in June 2024.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Shapes/SplitEqualizer.lean [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+/--
+The pair `f,g` is a `G`-cosplit pair if there is an `h : W ⟶ G X` so that `G f, G g, h` forms a
+split equalizer in `D`.
+-/
+abbrev Functor.IsCosplitPair : Prop :=
+  HasSplitEqualizer (G.map f) (G.map g)
+
+/-- Get the equalizer object from the typeclass `IsCosplitPair`. -/
+noncomputable def HasSplitEqualizer.equalizerOfCosplit [HasSplitEqualizer f g] : C :=
+  (@splittable _ _ _ _ f g).choose
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Limits/Shapes/SplitEqualizer.lean:137]: Can we get rid of this `@`? Similarly below, and in the other file.

File: Mathlib/CategoryTheory/Limits/Shapes/SplitEqualizer.lean [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+/--
+The pair `f,g` is a `G`-cosplit pair if there is an `h : W ⟶ G X` so that `G f, G g, h` forms a
+split equalizer in `D`.
+-/
+abbrev Functor.IsCosplitPair : Prop :=
+  HasSplitEqualizer (G.map f) (G.map g)
+
+/-- Get the equalizer object from the typeclass `IsCosplitPair`. -/
+noncomputable def HasSplitEqualizer.equalizerOfCosplit [HasSplitEqualizer f g] : C :=
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Limits/Shapes/SplitEqualizer.lean:136]: This def should be called `HasSplitEqualizer.equalizer` or something like that, as it has nothing to do with `CosplitPair`. I guess it's adapted from the other file, in which case that should be changed too.

File: Mathlib/CategoryTheory/Limits/Shapes/SplitEqualizer.lean (hunk 1 above)
Comment (dagurtomas) [reply Mathlib/CategoryTheory/Limits/Shapes/SplitEqualizer.lean]: I did this in both files.

File: Mathlib/CategoryTheory/Limits/Shapes/SplitEqualizer.lean (hunk 2 above)
Comment (dagurtomas) [reply Mathlib/CategoryTheory/Limits/Shapes/SplitEqualizer.lean]: Ok, I changed this name to `HasSplitEqualizer.equalizerOfSplit`, and didn't touch it in the other file. The `Split` in the name in the other file doesn't refer to `Functor.IsSplitPair`, hence we shouldn't use `Cosplit here.

**Review by dagurtomas — APPROVED** [review]

File: Mathlib/CategoryTheory/Limits/Shapes/SplitEqualizer.lean [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+satisfying `ι ≫ f = ι ≫ g` together with morphisms
+
+      r   t
+    W ← X ← Y
+
+satisfying `ι ≫ r = 𝟙 W`, `g ≫ t = 𝟙 X` and `f ≫ t = r ≫ ι`.
+
+The name "equalizer" is appropriate, since any split equalizer is a equalizer, see
+`Category_theory.IsSplitEqualizer.isEqualizer`.
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Limits/Shapes/SplitEqualizer.lean:54]:
```suggestion
`CategoryTheory.IsSplitEqualizer.isEqualizer`.
```
Maybe the other file as well?

**Comment by semorrison** [comment]
bors d+

**Comment by mckoen** [comment]
bors merge

**Comment by riccardobrasca** [comment]
What is the situation with this PR?

**Comment by mckoen** [comment]
> What is the situation with this PR?

I'm not sure, I thought it was okay to merge but it was cancelled for some reason. I'm not very familiar with Bors so maybe I did something wrong

**Comment by riccardobrasca** [comment]
Something failed, can you merge master and see what's wrong?

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**